cmake/compiler/gcc/target.cmake

```diff
@@ -42,21 +42,56 @@ execute_process(
     OUTPUT_VARIABLE temp_compiler_version
     )
 
-if("${temp_compiler_version}" VERSION_GREATER_EQUAL 13.1.0)
-    set(fix_header_file include/limits.h)
-else()
-    set(fix_header_file include-fixed/limits.h)
+set(HEADER_SEARCH_C_FLAGS "")
+if(CONFIG_PICOLIBC)
+  if (CONFIG_PICOLIBC_USE_MODULE)
+    list(APPEND NOSTDINC "${PROJECT_BINARY_DIR}/picolibc/include")
+  else()
+    set(HEADER_SEARCH_C_FLAGS "--specs=picolibc.specs")
+  endif()
 endif()
 
-foreach(file_name include/stddef.h "${fix_header_file}")
+foreach(_INCLUDE_LINE "#include <stddef.h>" "#include <limits.h>" "#include_next <limits.h>" "#include <stdbool.h>")
+
+  # Generate makefile dependencies for the target header file
   execute_process(
-    COMMAND ${CMAKE_C_COMPILER} --print-file-name=${file_name}
+    COMMAND echo "${_INCLUDE_LINE}"
+    COMMAND ${CMAKE_C_COMPILER} ${TOOLCHAIN_C_FLAGS} ${HEADER_SEARCH_C_FLAGS} -M -E -
```

Comment:
This doesn't seem to give the desired output on Windows.

CMake trace in Windows:
```
C:/projects/github/ncs/zephyr/cmake/compiler/gcc/target.cmake(54):  foreach(_INCLUDE_LINE #include <stddef.h> #include <limits.h> #include_next <limits.h> #include <stdbool.h> )
C:/projects/github/ncs/zephyr/cmake/compiler/gcc/target.cmake(57):  execute_process(COMMAND echo #include <stddef.h> COMMAND C:/opt/zephyr-sdk-0.16.1/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc.exe  --specs=picolibc.specs -M -E - OUTPUT_VARIABLE _OUTPUT )
C:/projects/github/ncs/zephyr/cmake/compiler/gcc/target.cmake(64):  string(REGEX REPLACE ^-:   _OUTPUT  )
...
```
as can be seen in the `target.cmake(64)` trace line, the content of `${_OUTPUT}` is empty.

In comparison to Linux:
```
/projects/github/ncs/zephyr/cmake/compiler/gcc/target.cmake(54):  foreach(_INCLUDE_LINE #include <stddef.h> #include <limits.h> #include_next <limits.h> #include <stdbool.h> )
/projects/github/ncs/zephyr/cmake/compiler/gcc/target.cmake(57):  execute_process(COMMAND echo #include <stddef.h> COMMAND /opt/zephyr-sdk-0.16.1/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc  --specs=picolibc.specs -M -E - OUTPUT_VARIABLE _OUTPUT )
/projects/github/ncs/zephyr/cmake/compiler/gcc/target.cmake(64):  string(REGEX REPLACE ^-:   _OUTPUT -: \
 /opt/zephyr-sdk-0.16.1/arm-zephyr-eabi/lib/gcc/arm-zephyr-eabi/12.2.0/include/stddef.h             
 )
...                                                                                            
```

Reply:
Can I get you to provide the contents of the `_OUTPUT` variable before any of the string operations? I don't have convenient access to a Windows system and will need help here.

Reply:
ok, maybe it didn't like the 'echo' command (which probably doesn't exist in Windows?). I've switched to using a temporary file; that works on Linux, maybe it will also work on Windows. I did take a look at the sdk compiler output for this case and it seems like the current regex "should" dtrt.

Reply:
> Can I get you to provide the contents of the _OUTPUT variable before any of the string operations?

it's empty, that's why this line: 
https://github.com/zephyrproject-rtos/zephyr/blob/12af5f338eabc561a483ddb8479d91ffb308e127/cmake/compiler/gcc/target.cmake#L64

expands to the following in the trace I included in first comment:
```
string(REGEX REPLACE ^-:   _OUTPUT  )
```
The `"${_OUTPUT}"` becomes empty, hence nothing is shown after `_OUTPUT`.